File: app/src/main/java/com/example/sarahn/waterreminderapp/customsViews/ViewPagerAdapter.java
package com.example.sarahn.waterreminderapp.customsViews;

import android.content.Context;
import android.graphics.Typeface;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.sarahn.waterreminderapp.R;

/**
 * Created by SarahN on 6/6/2017.
 */
public class ViewPagerAdapter extends PagerAdapter implements View.OnClickListener {

    LayoutInflater layoutInflater;
    int[] source;
    String[] title;
    Typeface typeface;
    Context context;


    public ViewPagerAdapter(Context context, int[] source) {
        this.context = context;
        this.source = source;
        layoutInflater = layoutInflater.from(context);
        typeface = Typeface.createFromAsset(context.getAssets(), "fonts/gb.otf");
        title = new String[]{"Get" +
                " periodic" +
                " reminder " +
                "to " +
                "drink water", "Drink" +
                " water " +
                "and " +
                "record", "Stay  hydrated!"};
    }

    @Override
    public int getCount() {
        return source.length;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view.equals(object);
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {

        View view = layoutInflater.inflate(R.layout.viewpager_layout, null);
        ImageView iv = (ImageView) view.findViewById(R.id.iv);
        TextView textView = (TextView) view.findViewById(R.id.tv);

        iv.setImageResource(source[position]);


        textView.setText(title[position]);
        textView.setTypeface(typeface);

        container.addView(view);

        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
           container.removeView((View)object);
    }


    @Override
    public void onClick(View v) {

    }


}

File: app/src/main/java/com/example/sarahn/waterreminderapp/customsViews/ViewPagerAdapter.kt
package com.example.sarahn.waterreminderapp.customsViews

import android.content.Context
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.viewpager.widget.PagerAdapter
import com.example.sarahn.waterreminderapp.R

/**
 * Created by SarahN on 6/6/2017.
 */
class ViewPagerAdapter(var context: Context, var source: IntArray) : PagerAdapter(), View.OnClickListener {
    var layoutInflater: LayoutInflater
    var title: Array<String>
    var typeface: Typeface
    override fun getCount(): Int {
        return source.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = layoutInflater.inflate(R.layout.viewpager_layout, null)
        val iv = view.findViewById<View>(R.id.iv) as ImageView
        val textView = view.findViewById<View>(R.id.tv) as TextView
        iv.setImageResource(source[position])
        textView.text = title[position]
        textView.typeface = typeface
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

    override fun onClick(v: View) {}

    init {
        layoutInflater = LayoutInflater.from(context)
        typeface = Typeface.createFromAsset(context.assets, "fonts/gb.otf")
        title = arrayOf("Get" +
                " periodic" +
                " reminder " +
                "to " +
                "drink water", "Drink" +
                " water " +
                "and " +
                "record", "Stay  hydrated!")
    }
}